### ISSUE TITLE
AssertionError: Failed to open youtube link

### ISSUE THREAD

**Issue by PhilPark-geosr** [ultralytics/yolov5#3016]
Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following, otherwise it is non-actionable, and we can not help you:
 - **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
 - **Common dataset**: coco.yaml or coco128.yaml
 - **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#environments
 
If this is a custom dataset/training question you **must include** your `train*.jpg`, `test*.jpg` and `results.png` figures, or we can not help you. You can generate these with `utils.plot_results()`.

## To Reproduce (REQUIRED)

Input:
python detect.py --source 'http://youtu.be/gmklhuYYL1k'

Output:
```
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, img_size=640, iou_thres=0.45, name='exp', project='runs/detect', save_conf=False, save_txt=False, source='https://youtu.be/gmklhuYYL1k', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 v4.0-16-g777032e torch 1.8.1+cu102 CUDA:0 (GeForce RTX 2080 Ti, 11019.4375MB)
                                          CUDA:1 (GeForce RTX 2080 Ti, 11019.4375MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients, 17.0 GFLOPS
Traceback (most recent call last):
  File "detect.py", line 173, in <module>
    detect()
  File "detect.py", line 52, in detect
    dataset = LoadImages(source, img_size=imgsz)
  File "/home/ppark/yolo/yolov5/utils/datasets.py", line 132, in __init__
    raise Exception(f'ERROR: {p} does not exist')
Exception: ERROR: /home/ppark/yolo/yolov5/https:/youtu.be/gmklhuYYL1k does not exist
(yolov5) ppark@geosr:~/yolo/yolov5$ python detect.py --source 'http://youtu.be/gmklhuYYL1k'
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, img_size=640, iou_thres=0.45, name='exp', project='runs/detect', save_conf=False, save_txt=False, source='http://youtu.be/gmklhuYYL1k', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 v4.0-16-g777032e torch 1.8.1+cu102 CUDA:0 (GeForce RTX 2080 Ti, 11019.4375MB)
                                          CUDA:1 (GeForce RTX 2080 Ti, 11019.4375MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients, 17.0 GFLOPS
1/1: http://youtu.be/gmklhuYYL1k... Traceback (most recent call last):
  File "detect.py", line 173, in <module>
    detect()
  File "detect.py", line 49, in detect
    dataset = LoadStreams(source, img_size=imgsz)
  File "/home/ppark/yolo/yolov5/utils/datasets.py", line 276, in __init__
    assert cap.isOpened(), f'Failed to open {s}'
AssertionError: Failed to open http://youtu.be/gmklhuYYL1k
```

## Environment

 - OS: [e.g. Ubuntu]
 - GPU [e.g. 2080 Ti]

**Comment by iceisfun** [comment]
Have you tried something like youtube-dl to download the video then process it with detect?

the detect.py script does not integrate with things like youtube to download content.

Try something like this

```
root@21dc4c9707f0:/usr/src/app# pip3 install youtube-dl
Collecting youtube-dl
  Downloading youtube_dl-2021.4.26-py2.py3-none-any.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB 2.7 MB/s 
Installing collected packages: youtube-dl
Successfully installed youtube-dl-2021.4.26
root@21dc4c9707f0:/usr/src/app# youtube-dl https://www.youtube.com/watch?v=gmklhuYYL1k
[youtube] gmklhuYYL1k: Downloading webpage
[youtube] gmklhuYYL1k: Downloading MPD manifest
[download] Destination: 유하영&옥성결-gmklhuYYL1k.mp4
[download] 100% of 103.71MiB in 00:16
root@21dc4c9707f0:/usr/src/app# python detect.py --source ./유하영\&옥성결-gmklhuYYL1k.mp4 
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, hide_conf=False, hide_labels=False, img_size=640, iou_thres=0.45, line_thickness=3, name='exp', nosave=False, project='runs/detect', save_conf=False, save_crop=False, save_txt=False, source='./유하영&옥성결-gmklhuYYL1k.mp4', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 🚀 v5.0-59-g31ee54c torch 1.8.0a0+52ea372 CUDA:0 (TITAN RTX, 24220.3125MB)
                                                CUDA:1 (TITAN RTX, 24220.3125MB)
                                                CUDA:2 (TITAN RTX, 24220.3125MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients
video 1/1 (1/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.017s)
video 1/1 (2/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (3/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 1 couch, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (4/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 7 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (5/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (6/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (7/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (8/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 6 persons, 4 chairs, 1 couch, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (9/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 6 persons, 3 chairs, 2 potted plants, 1 tv, 3 laptops, Done. (0.009s)
video 1/1 (10/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 6 persons, 3 chairs, 2 potted plants, 1 tv, 2 laptops, Done. (0.009s)
video 1/1 (11/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 7 persons, 5 chairs, 2 potted plants, 2 laptops, Done. (0.009s)
video 1/1 (12/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 8 persons, 5 chairs, 1 couch, 2 potted plants, 2 laptops, Done. (0.009s)
video 1/1 (13/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 6 persons, 5 chairs, 3 potted plants, 2 laptops, Done. (0.009s)
video 1/1 (14/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 7 persons, 5 chairs, 3 potted plants, 2 laptops, Done. (0.009s)
video 1/1 (15/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 6 persons, 4 chairs, 3 potted plants, 2 laptops, Done. (0.009s)
video 1/1 (16/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 5 persons, 4 chairs, 3 potted plants, 1 laptop, Done. (0.009s)
video 1/1 (17/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 4 persons, 3 chairs, 3 potted plants, 3 laptops, Done. (0.009s)
video 1/1 (18/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 3 persons, 3 chairs, 3 potted plants, 3 laptops, Done. (0.009s)
video 1/1 (19/11987) /usr/src/app/유하영&옥성결-gmklhuYYL1k.mp4: 384x640 2 persons, 3 chairs, 3 potted plants, 1 laptop, Done. (0.009s)
```

**Comment by pravastacaraka** [comment]
@PhilPark-geosr update the yolov5 repo with `git pull` to get the latest version which is v5.0. There is a new feature added in PR #2752, inference using the youtube link.

**Comment by glenn-jocher** [comment]
@PhilPark-geosr YouTube inference is working correctly. If I run your command in a local PyCharm terminal:

```shell
(venv) (base) glennjocher@Glenns-iMac yolov5 % python detect.py --source 'http://youtu.be/gmklhuYYL1k' 
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, hide_conf=False, hide_labels=False, img_size=640, iou_thres=0.45, line_thickness=3, name='exp', nosave=False, project='runs/detect', save_conf=False, save_crop=False, save_txt=False, source='http://youtu.be/gmklhuYYL1k', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 🚀 v5.0-60-ge2a80c6 torch 1.8.1 CPU

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients
1/1: http://youtu.be/gmklhuYYL1k...  success (1280x720 at 29.97 FPS).

0: 384x640 5 persons, 3 chairs, 2 potted plants, 1 tv, 1 laptop, Done. (0.224s)
0: 384x640 6 persons, 3 chairs, 2 potted plants, 1 tv, 3 laptops, Done. (0.204s)
0: 384x640 6 persons, 5 chairs, 3 potted plants, 2 laptops, Done. (0.210s)
0: 384x640 3 persons, 3 chairs, 3 potted plants, 1 laptop, Done. (0.202s)
0: 384x640 3 persons, 4 chairs, 2 potted plants, 2 laptops, Done. (0.237s)
```

<img width="1280" alt="Screenshot 2021-05-03 at 12 06 27" src="https://user-images.githubusercontent.com/26833433/116864362-05aef500-ac08-11eb-900d-519610820ead.png">

**Comment by glenn-jocher** [comment]
@PhilPark-geosr good news 😃! A related issue to your original issue may now been fixed ✅ in #3018 allowing YouTube inference in Google Colab instances.  To receive this update you can:
- `git pull` from within your `yolov5/` directory
- `git clone https://github.com/ultralytics/yolov5` again
- Force-reload [PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/): `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- View our updated [Colab](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) and [Kaggle](https://www.kaggle.com/ultralytics/yolov5-ultralytics) notebooks

Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

<img width="1184" alt="Screenshot 2021-05-03 at 12 38 59" src="https://user-images.githubusercontent.com/26833433/116866988-8cfe6780-ac0c-11eb-824b-e6107f8fa424.png">
